docs/DeveloperGuide.md

```diff
@@ -1,29 +1,240 @@
 # Developer Guide
```

Comment:
Good use of class diagram and sequence diagram, would it be possible for you to add the object diagram as well?